stm/src/transaction.rs

```diff
@@ -0,0 +1,299 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{ctrl::MemoryController, BoxedMemory, TLog, TVar, TransactionError};
+// use lazy_static::*;
+use log::*;
+use std::{
+    collections::{btree_map::Entry, BTreeMap},
+    future::Future,
+    pin::Pin,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    task::{Context, Poll},
+};
+
+/// Defines a transaction strategy. Strategies vary
+/// how transactions should be handled in case the commit
+/// to memory fails.
+#[derive(Clone, Copy)]
+pub enum Strategy {
```

Comment:
I like the idea of an exposed strategy API.  Makes sense in the context of the Stronghold runtime.  I wonder if its worth adding a strategy for dealing with runtime "DDOSing" (lack of a better term) where a user calls a procedure in perpetuity so they can attempt to dump the memory while the secrets are revealed.  Would it make sense to add that here or make it a procedure level API?

Reply:
That's actually a good idea to tackle this at stm level, having some kind of execution throttling. This could be done by queuing up a limited number of transactions, blocking any further requests,